### PR TITLE
Fixes #866 should not throw NPE, when custom Exception fillInStackTrace returns null

### LINKED ISSUE #866: When custom exception fillInstackTrace() returns null, Exception mock cannot work properly
```java
public class CustomException extends Exception {

    public CustomException() {
        super();
    }

    @Override
    public Exception fillInStackTrace() {
        return null;
    }
}
```

TestCase like this
```java
 when(demo.process()).thenThrow(new CustomException())

//... test code
```

```
java.lang.NullPointerException
	at org.mockito.internal.exceptions.stacktrace.ConditionalStackTraceFilter.filter(ConditionalStackTraceFilter.java:23)
	at org.mockito.internal.stubbing.answers.ThrowsException.answer(ThrowsException.java:32)
	at org.mockito.internal.stubbing.StubbedInvocationMatcher.answer(StubbedInvocationMatcher.java:35)
	at org.mockito.internal.handler.MockHandlerImpl.handle(MockHandlerImpl.java:95)
```

### PR DESCRIPTION
**Pull request by wuwen5**
fixes #866.

### CODE REVIEW

File: src/main/java/org/mockito/internal/stubbing/answers/ThrowsException.java [hunk 1]
```diff
@@ -29,6 +29,10 @@ public Object answer(InvocationOnMock invocation) throws Throwable {
             throw throwable;
         }
         Throwable t = throwable.fillInStackTrace();
+
+        if (t == null) {
+            throw throwable;
+        }
```
Comment (bric3) [inline src/main/java/org/mockito/internal/stubbing/answers/ThrowsException.java]: Actually I'd rather make the check in `ConditionalStackTraceFilter.filter()`. And if throwable is null, just return.

Comment (wuwen5) [inline src/main/java/org/mockito/internal/stubbing/answers/ThrowsException.java]: 👌 already changed,please review.

**Comment by bric3** [comment]
The build is failing.

File: src/main/java/org/mockito/internal/stubbing/answers/ThrowsException.java [hunk 2]
```diff
@@ -30,7 +30,7 @@ public Object answer(InvocationOnMock invocation) throws Throwable {
         }
         Throwable t = throwable.fillInStackTrace();
         filter.filter(t);
-        throw t;
+        throw throwable;
```
Comment (bric3) [inline src/main/java/org/mockito/internal/stubbing/answers/ThrowsException.java]: Sorry it seems I was wrong. It seems that indeed your previous attempt was correct.

```java
        if (t == null) {
            return throwable;
        }
```

Comment (wuwen5) [inline src/main/java/org/mockito/internal/stubbing/answers/ThrowsException.java]: Ok, But I don't know why the build is failing?

> Execution failed for task ':assertEnvVariables'.
> Please export 'GH_TOKEN' env variable first. Expression: java.lang.System.env[it]. Values: it = GH_TOKEN

Comment (bric3) [inline src/main/java/org/mockito/internal/stubbing/answers/ThrowsException.java]: Yes I'm sure your change is correct, it's something else.

File: src/main/java/org/mockito/internal/stubbing/answers/ThrowsException.java (hunk 1 above)
Comment (bric3) [inline src/main/java/org/mockito/internal/stubbing/answers/ThrowsException.java]: Discarding this comment as it requires to rewrite `filter` a bit more.

**Pull request merged**